cmd/mavenBuild.go

```diff
@@ -17,16 +19,34 @@ import (
 	piperhttp "github.com/SAP/jenkins-library/pkg/http"
 )
 
-func mavenBuild(config mavenBuildOptions, telemetryData *telemetry.CustomData) {
+// ToDo : move the struct to a slim new buildSettings Package in future
+type BuildSettingsInfo struct {
+	Profiles                    []string `json:"profiles,omitempty"`
+	Publish                     bool     `json:"publish,omitempty"`
+	CreateBOM                   bool     `json:"createBOM,omitempty"`
+	LogSuccessfulMavenTransfers bool     `json:"logSuccessfulMavenTransfers,omitempty"`
+	GlobalSettingsFile          string   `json:"globalSettingsFile,omitempty"`
+	DefaultNpmRegistry          string   `json:"defaultNpmRegistry,omitempty"`
+}
+
+// ToDo : move the struct to a slim new buildSettings Package in future
+type BuildSettings struct {
+	MavenBuild  []BuildSettingsInfo `json:"mavenBuild,omitempty"`
+	NpmBuild    []BuildSettingsInfo `json:"npmBuild,omitempty"`
+	DockerBuild []BuildSettingsInfo `json:"dockerBuild,omitempty"`
```

Comment:
shouldnt't we rather use the names of the build steps that we have a proper 1:1 mapping?

We may have different types on how we execute a Docker build (Kaniko, CNB, Docker native, ...)